src/main/java/seedu/duke/validation/Validation.java

```diff
@@ -0,0 +1,203 @@
+     */
+    public static void ensureNonEmptyCommand(String command) throws DukeException {
+        if (command.equals("")) {
+            throw new DukeException("Command cannot be empty.");
+        }
+    }
+
+    /**
+     * Returns a validated Date object.
+     * Ensures that input String is in valid DD/MM/YYYY format.
+     * Ensures there are at most 12 months in a year.
+     * Ensures the day of the month in the year entered is valid.
+     * @param date String representation of a date in DD/MM/YYYY format.
+     * @return Validated Date object containing the day, month, year of the date parameter.
+     * @throws DukeException If the date entered is invalid.
+     */
+    public static Date getValidatedDate(String date) throws DukeException {
+        if (date.length() != 10) {
+            throw new DukeException("Please ensure date format is in DD/MM/YYYY");
+        }
+
+        String[] dateArr = date.split("/");
+        if (dateArr.length != 3) {
+            throw new DukeException("Please ensure date format is in DD/MM/YYYY");
+        }
+
+        int day = 0;
+        int month = 0;
+        int year = 0;
+        boolean isLeapYear = false;
+
+        try {
+            day = Integer.parseInt(dateArr[0]);
+            month = Integer.parseInt(dateArr[1]);
+            year = Integer.parseInt(dateArr[2]);
+            if (day < 1) {
+                throw new DukeException("Calendar day must start from 1");
+            } else if (month < 1) {
+                throw new DukeException("Calendar month must start from 1");
+            } else if (year < 0) {
+                throw new DukeException("Calendar year must start from 0");
+            } else if (month > 12) {
+                throw new DukeException("There cannot be more than 12 months in a year.");
+            }
+        } catch (NumberFormatException ex) {
+            throw new DukeException("Please ensure date format is in DD/MM/YYYY");
+        }
+
+        if (year % 4 == 0) {
+            isLeapYear = true;
+        } else if (year % 100 == 0 && year % 400 == 0) {
+            isLeapYear = true;
+        }
+
+        if ((month <= 7 && month % 2 != 0) || (month > 7 && month % 2 == 0)) {
+            if (day > 31) {
+                throw new DukeException("There cannot be more than 31 days in " + Date.getMonth(month));
+            }
+        } else if (month != 2) {
+            if (day > 30) {
+                throw new DukeException("There cannot be more than 30 days in " + Date.getMonth(month));
+            }
+        } else {
+            if (isLeapYear && day > 29) {
+                throw new DukeException("There cannot be more than 29 days in " + Date.getMonth(month) + " "
+                        + year + ".");
+            } else if (!isLeapYear && day > 28) {
+                throw new DukeException("There cannot be more than 28 days in " + Date.getMonth(month) + " "
+                        + year + ".");
+            }
+        }
+
```

Comment:
Very detailed and accurate information shown to user!

Reply:
Thanks for the positive feedback! Glad to know it's helpful to the user :)